File: hrms-backend/src/main/java/com/finalproject/hrmsbackend/business/abstracts/ImageService.java
package com.finalproject.hrmsbackend.business.abstracts;

import com.finalproject.hrmsbackend.core.utilities.results.DataResult;
import com.finalproject.hrmsbackend.core.utilities.results.Result;
import com.finalproject.hrmsbackend.entities.concretes.Image;
import org.springframework.web.multipart.MultipartFile;

public interface ImageService {

    DataResult<Image> getById(int imgId);

    DataResult<Image> getByPublicId(String publicId);

    Result upload(MultipartFile multipartFile, int userId);

    Result deleteById(int imgId);

    Result deleteByPublicId(String publicId);

}

File: hrms-frontend/src/services/imageService.js
import axios from "axios";

export default class ImageService {

    getById(imgId) {
        return axios.get(`http://localhost:8080/api/images/get/byId?imgId=${imgId}`)
    }

    upload(userId, multipart) {
        return axios.post(`http://localhost:8080/api/images/upload?userId=${userId}`, multipart)
    }

    deleteById(imgId) {
        return axios.delete(`http://localhost:8080/api/images/delete/byId?imgId=${imgId}`)
    }

}